pulp_2to3_migration/app/plugin/rpm/pulp_2to3_models.py

```diff
@@ -524,3 +536,236 @@ class Meta:
+        return PackageLangpacks(matches=langpacks_dict['matches'],
+                                digest=dict_digest(langpacks_dict))
+
+
+class Pulp2PackageGroup(Pulp2to3Content):
+    """
+    Pulp 2to3 detail content model to store pulp 2 package_group
+    content details for Pulp 3 content creation.
+    """
+    package_group_id = models.TextField()
+    repo_id = models.TextField()
+
+    default = models.BooleanField()
+    user_visible = models.BooleanField()
+    display_order = models.IntegerField(null=True)
+    name = models.TextField()
+    description = models.TextField()
+
+    # merge optional, conditional, default, mandatory
+    packages = JSONField()
+    desc_by_lang = JSONField(dict)
+    name_by_lang = JSONField(dict)
+    biarch_only = models.BooleanField(default=False)
+
+    pulp2_type = 'package_group'
+
+    class Meta:
+        unique_together = ('repo_id', 'package_group_id', 'pulp2content')
+        default_related_name = 'package_group_detail_model'
+
+    @classmethod
+    async def pre_migrate_content_detail(cls, content_batch):
+        """
+        Pre-migrate Pulp 2 package groups with all the fields needed to create a Pulp 3 content.
+
+        Args:
+             content_batch(list of Pulp2Content): pre-migrated generic data for Pulp 2 content.
+
+        """
+        def _get_packages(group):
+            """Merge in a list all package types"""
+            return [(0, group.default_package_names), (1, group.optional_package_names),
+                    (3, group.mandatory_package_names), (2, group.conditional_package_names)]
+
+        pulp2_id_obj_map = {pulp2content.pulp2_id: pulp2content for pulp2content in content_batch}
+        pulp2_ids = pulp2_id_obj_map.keys()
+        pulp2_content_batch = pulp2_models.PackageGroup.objects.filter(id__in=pulp2_ids)
+        pulp2groups_to_save = [
+            cls(repo_id=p.repo_id,
+                package_group_id=p.package_group_id,
+                default=p.default,
+                user_visible=p.user_visible,
+                display_order=p.display_order,
+                name=p.name,
+                description=p.description if p.description else '',
+                packages=_get_packages(p),
+                desc_by_lang=p.translated_description,
+                name_by_lang=p.translated_name,
+                pulp2content=pulp2_id_obj_map[p.id])
+            for p in pulp2_content_batch]
+        cls.objects.bulk_create(pulp2groups_to_save, ignore_conflicts=True)
+
+    async def create_pulp3_content(self):
+        """
+        Create a Pulp 3 Package Group content for saving it later in a bulk operation.
+        """
+        group = pkg_grp_to_libcomps(self)
+        group_dict = PackageGroup.libcomps_to_dict(group)
+        packages = group_dict['packages']
+        # ugly stuff
+        for pkg in packages:
+            if pkg['requires'] == '':
+                pkg['requires'] = None
```

Comment:
`pkg['requires'] = pkg['requires'] or None` ?

Reply:
This would work, +1